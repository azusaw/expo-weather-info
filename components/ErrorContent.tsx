import { Button, StyleSheet } from "react-native";
import { Text, View } from "@/components/Themed";
import Colors from "@/constants/Colors";

const ErrorContent = ({
  message,
  onRetry,
}: {
  message: string;
  onRetry: () => void;
}) => (
  <View>
    <Text size={42} weight={700} color={Colors.text.light}>
      Sorry...
    </Text>
    <Text size={22} color={Colors.text.light} style={styles.description}>
      {message}
    </Text>
    <Button title="Retry" onPress={onRetry} color={Colors.primary.dark} />
  </View>
);

const styles = StyleSheet.create({
  description: {
    marginTop: 20,
    marginBottom: 40,
  },
});

export default ErrorContent;
