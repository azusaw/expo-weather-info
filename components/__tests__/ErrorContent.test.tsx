import React from "react";
import { fireEvent, render, waitFor } from "@testing-library/react-native";
import ErrorContent from "@/components/ErrorContent";

const mockProps = {
  message: "Something went wrong while getting weather data.",
  onRetry: jest.fn,
};

describe("ErrorContent", () => {
  test("it renders ErrorContent successfully", () => {
    const { toJSON } = render(<ErrorContent {...mockProps} />);
    expect(toJSON()).toMatchSnapshot();
  });

  test("it renders message", () => {
    const { getByText } = render(<ErrorContent {...mockProps} />);
    expect(getByText(mockProps.message)).toBeTruthy();
  });

  test("it calls onRetry when button is clicked", () => {
    const { getByRole } = render(<ErrorContent {...mockProps} />);
    const target = getByRole("button");
    fireEvent.press(target);
    waitFor(() => expect(mockProps.onRetry).toHaveBeenCalled());
  });
});
