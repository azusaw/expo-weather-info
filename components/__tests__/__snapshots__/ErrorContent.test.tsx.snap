// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorContent it calls onRetry when button is clicked 1`] = `
<View
  style={
    [
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#eee",
          "fontFamily": "Montserrat_700Bold",
          "fontSize": 42,
        },
        undefined,
      ]
    }
  >
    Sorry...
  </Text>
  <Text
    style={
      [
        {
          "color": "#eee",
          "fontFamily": "Montserrat_400Regular",
          "fontSize": 22,
        },
        {
          "marginBottom": 40,
          "marginTop": 20,
        },
      ]
    }
  >
    Something went wrong while getting weather data.
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            {
              "color": "#445275",
            },
          ]
        }
      >
        Retry
      </Text>
    </View>
  </View>
</View>
`;

exports[`ErrorContent it renders ErrorContent successfully 1`] = `
<View
  style={
    [
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#eee",
          "fontFamily": "Montserrat_700Bold",
          "fontSize": 42,
        },
        undefined,
      ]
    }
  >
    Sorry...
  </Text>
  <Text
    style={
      [
        {
          "color": "#eee",
          "fontFamily": "Montserrat_400Regular",
          "fontSize": 22,
        },
        {
          "marginBottom": 40,
          "marginTop": 20,
        },
      ]
    }
  >
    Something went wrong while getting weather data.
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            {
              "color": "#445275",
            },
          ]
        }
      >
        Retry
      </Text>
    </View>
  </View>
</View>
`;

exports[`ErrorContent it renders SvgIcon successfully 1`] = `
<View
  style={
    [
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#eee",
          "fontFamily": "Montserrat_700Bold",
          "fontSize": 42,
        },
        undefined,
      ]
    }
  >
    Sorry...
  </Text>
  <Text
    style={
      [
        {
          "color": "#eee",
          "fontFamily": "Montserrat_400Regular",
          "fontSize": 22,
        },
        {
          "marginBottom": 40,
          "marginTop": 20,
        },
      ]
    }
  >
    Something went wrong while getting weather data.
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            {
              "color": "#445275",
            },
          ]
        }
      >
        Retry
      </Text>
    </View>
  </View>
</View>
`;

exports[`ErrorContent it renders SvgIcon successfully 2`] = `
<View
  style={
    [
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#eee",
          "fontFamily": "Montserrat_700Bold",
          "fontSize": 42,
        },
        undefined,
      ]
    }
  >
    Sorry...
  </Text>
  <Text
    style={
      [
        {
          "color": "#eee",
          "fontFamily": "Montserrat_400Regular",
          "fontSize": 22,
        },
        {
          "marginBottom": 40,
          "marginTop": 20,
        },
      ]
    }
  >
    Something went wrong while getting weather data.
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            {
              "color": "#445275",
            },
          ]
        }
      >
        Retry
      </Text>
    </View>
  </View>
</View>
`;
